### PR TITLE
Jc/switch to runtime classpath

### LINKED ISSUE #201: gradle 3.4 java-library support using runtimeClasspath
> When resolving the runtime classpath for Java applications, Gradle will now use the runtimeClasspath configuration instead of the runtime configuration. If you previously attached resolution rules to runtime, you should apply them to runtimeClasspath instead or apply them to all configurations.

in Gradle 3.4 ([release notes][1]), they've moved to using the `runtimeClasspath` as the configuration rather than `runtime`. i believe this means enhancing this [bit of code][2] in the `DistTarTask` to include the `runtimeClasspath`.

i believe you can check if the configuration exists by name and then add it conditionally, which should let it work with pas versions of gradle.

would a PR for this make sense?

[1]: https://docs.gradle.org/3.4/release-notes.html#runtimeclasspath-used-instead-of-runtime
[2]: https://github.com/palantir/sls-packaging/blob/develop/gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/tasks/DistTarTask.groovy#L61-L64

### PR DESCRIPTION
**Pull request by jmcampanini**
fixes #201

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/sls-packaging/205)

### CODE REVIEW

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy [hunk 1]
```diff
@@ -497,8 +497,10 @@ class ServiceDistributionPluginTests extends GradleTestSpec {
         def classpathJar = file('dist/service-name-0.0.1/service/lib/').listFiles()
                 .find({ it.name.endsWith("-manifest-classpath-0.0.1.jar") })
         classpathJar.exists()
-        readFromZip(classpathJar, "META-INF/MANIFEST.MF")
-                .contains('Class-Path: guava-19.0.jar produces-manifest-') // etc
+        String fromZip = readFromZip(classpathJar, "META-INF/MANIFEST.MF")
+        fromZip.contains('Class-Path:')
+        fromZip.contains('guava-19.0.jar')
+        fromZip.contains('produces-manifest-')
```
Comment (uschi2000) [inline gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy]: why did this change?

Comment (jmcampanini) [reply gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy]: the old version required that `guava-19.0.jar` was the first entry:

```
 .contains('Class-Path: guava-19.0.jar produces-manifest-')
```

when using `runtimeClasspath` it changed to say `main main guava-19.0.jar`, which i assume means that the main directories were included in the classpath.

which is technically correct that it's the full classpath, but if you were expecting to only have the JAR files, then we'd need to filter the file collection to remove certain entities.

Comment (uschi2000) [reply gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy]: but there is no `main` directory in the distribution, right? it's just the jars.

**Comment by jmcampanini** [comment]
@uschi2000 updated it to find all configurations found within the classpath. this should:

1. mimic the existing behavior by enumerating only configurations found in the `runtimeClasspath`
1. leave the tests the same
1. remove the `main` folders we were seeing
1. work with a `java-library` example

also added a test to ensure it works. lemme know if there's anything else necessary or if this isn't a good approach

**Comment by jmcampanini** [comment]
silly me, there is a configuration called `runtimeClasspath`, no need to pull it from sourceSets. updated

**Review by uschi2000 — APPROVED** [review]

**Comment by uschi2000** [comment]
This looks much better, thanks Javier!

**Pull request merged**